packages/components/src/snackbar/index.tsx

```diff
@@ -19,7 +19,7 @@ import Button from '../button';
 import type { NoticeAction, SnackbarProps } from './types';
 import type { WordPressComponentProps } from '../ui/context';
 
-const NOTICE_TIMEOUT = 10000;
+const NOTICE_TIMEOUT = 5000;
```

Comment:
Often snackbar notices might have a link in them that can be interacted with, e.g. if you save a draft, you can click to view the preview:

<img width="270" alt="image" src="https://user-images.githubusercontent.com/14988353/212779100-47b15419-3e14-45bc-b507-b3d93b98d5f9.png">

Would globally reducing the timeout make it harder for these interactive ones, or snackbars that contain more descriptive info? I'm mostly thinking of trying to ensure that users have enough time to read the snackbar, then make a decision about whether to interact with it, and then move the mouse over to the button and click on it. Would introducing a configurable timeout value help, so we could potentially preserve the component's default value of 10s, but then define a shorter timeout in specific usage?

Reply:
Ah yes, this is a good point! I don't think 5 seconds is long enough for interactive notices. I'd imagine also the reason they are 10 seconds. I think a configurable, or at least a context aware timeout is a better idea. 👍

Reply:
One issue with making it configurable is that all of the timeouts will inevitably be inconsistent.

It does seem from looking into the history of this (https://github.com/WordPress/gutenberg/pull/15594#issuecomment-492619316) that 10s was fairly intentional:

> Snackbar should disappear after a relatively longer period of time to allow people time to interact with them (10s) and should also be dismissable (even if via a non-visible action, like tapping on the snackbar itself) so as to ensure it doesn't get in users' way and allow for more choice.

Reply:
How about 6 seconds as a starting point? I'd like to think that if the main purpose of the long delay is the "View preview" link in the snackbar, we can take that as a prompt to improve this area, which has a permanent link to it:

<img width="385" alt="Screenshot 2023-01-17 at 09 40 32" src="https://user-images.githubusercontent.com/1204802/212849765-1c7ff9ab-e10d-424f-bc53-c9a9ce8e88d0.png">

Reply:
@jasmussen I agree that sidebar view could improve, I'd actually be interested in working on that. The trouble is you don't see this when saving a draft of the post (when you get the View Preview snackbar). So it doesn't help in removing the need for it.